src/Mvc/Mvc.Core/src/ModelBinding/Binders/DictionaryModelBinder.cs

```diff
@@ -126,6 +126,11 @@ public override async Task BindModelAsync(ModelBindingContext bindingContext)
         await base.BindModelAsync(bindingContext);
         if (!bindingContext.Result.IsModelSet)
         {
+            if (bindingContext.IsTopLevelObject)
```

Comment:
What scenario does this address?

Reply:
It is very weird, because if you are setting a default value should means Binding is not required but this is the scenario:

``` c#
Get([BindRequired] Dictionary<string, string> test = null)
```

Reply:
Please add a comment explaining why it is needed for future devs